### ISSUE TITLE
Immortals Fenyx Rising (2221920)

### ISSUE THREAD

**Issue by Bitwolfies** [ValveSoftware/Proton#6393]
# Compatibility Report
- Name of the game with compatibility issues: Immortals Fenyx Rising
- Steam AppID of the game: 2221920

## System Information
-Steam Deck 
- Proton version: Experimental bleeding edge

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms 

Audio massively desyncs and repeats during any cutscene.
## Reproduction
Just start the game and wait for the first cutscene to start messing up its audio.

**Comment by ffrasisti** [comment]
Im also on Steam Deck. Latest stable, 64gb. 

I have a couple more
## **Symptoms**
> Flickering in the grass in daylight in the overworld
Yellow grass starts flickering slightly with broken black shading.

> Game freezes when deck goes into sleep mode


Proton Hotfix makes the audio issue better as parts of cinematics run fine but other parts start REPEATING dialogue or skipping lines, as if it doesnt know how to read the game's dialogue and changes parts of the cutscene around or skips some. It's really weird. The very first part in Hotfix runs fine until we get to Zeus and Prometheus. I also tested Experimental and 7.0.5 and it's the same.  **Curiously, according to youtubers testing the game on deck months before it came to steam, the versions of the game in other stores dont seem to have this problem, so there may be something specific to the steam version here**. 

For reference here's the steam version and its issues:
https://www.youtube.com/watch?v=mb1ykoXyuJg&t=1s

And here's one from another launcher months ago:
https://www.youtube.com/watch?v=xHVXdcrwVZk It's not the same cutscene exactly, but the cutscenes dont exhibit the same problem here.
Hopefully it will be fixed soon.

**Comment by ffrasisti** [comment]
I can also confirm everything op said, obviously. Im on the latest stable steamos. 64gb model, game running from a 1tb sandisk extreme SD

**Comment by Bitwolfies** [comment]
I will also attest to the flickering grass, but I don’t personally think the audio issue is  specific to the steam version as I can find threads from 2 years ago complaining about the same issue.

**Comment by Blisto91** [comment]
The audio issue have been fixed on bleeding edge and be pulled into regular experimental in a future update.

The grass issue looks like a driver problem and I'll look into making a mesa issue this weekend.

Edit: oh wait. Do you have the audio issue on latest bleeding edge?

**Comment by Bitwolfies** [comment]
> The audio issue have been fixed on bleeding edge and be pulled into regular experimental in a future update.
> 
> The grass issue looks like a driver problem and I'll look into making a mesa issue this weekend.
> 
> Edit: oh wait. Do you have the audio issue on latest bleeding edge?

On my deck yes, I swapped the beta to bleeding edge before making the report, so it should be up to date. Assuming this audio patch was not literally published after making my report.

**Comment by Blisto91** [comment]
No it was released 13 hours ago.
Can you share a proton log? :slightly_smiling_face: 

Edit: okay i see. Theres is still a issue on Deck for some reason even tho it is fixed on my desktop where i tested.

**Comment by Blisto91** [comment]
Okay yes the issue doesn't seem to have been fixed fully, only improved a bit compared to before the fix. Seems to depend on the hardware and graphical settings.

**Comment by ffrasisti** [comment]
The issue apparently is fixed by this on a different game, AC Odyssey. something here affects _only the steam version_ of this game. Can someone from proton take a look please?

**Comment by ffrasisti** [comment]
> The audio issue have been fixed on bleeding edge and be pulled into regular experimental in a future update.
> 
> The grass issue looks like a driver problem and I'll look into making a mesa issue this weekend.
> 
> Edit: oh wait. Do you have the audio issue on latest bleeding edge?

did you make the mesa report?

**Comment by Blisto91** [comment]
> The issue apparently is fixed by this on a different game, AC Odyssey. something here affects only the steam version of this game. Can someone from proton take a look please?

They are aware of the issue.


> did you make the mesa report?

Not yet. Will try to make it today. The apitraces i wanted to use for easy reproduction are not cooperating.

Edit: issue made

**Comment by ffrasisti** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6393#issuecomment-1356760501

Thanks. Hopefully someone will look into this audio jumping problem.

**Comment by Bitwolfies** [comment]
Newest build of GE proton claims to have fixed the issue, haven’t tested it but it’s worth noting. Wonder if it’s the same fix in this thread that only mitigated it or a new proper one.

**Comment by Blisto91** [comment]
It's the same as bleeding edge from what I can tell from a quick look.

**Comment by Bitwolfies** [comment]
> It's the same as bleeding edge from what I can tell from a quick look.

If that’s the case that’s very odd cause I’m testing it now, and it has fixed the issue.

**Comment by Blisto91** [comment]
On Steam Deck?

Fwiw it's the games audio engine that is a bit weird. You can make this happen with the native game on Windows, but you have to turn everything pretty overkill up compared to what your machine can handle, so it probably isn't a common problem there. The proton overhead or other differences seems to make it worse or more common.
Or if you ntfs compress the games folder on Windows then it can happen too. 
But ofc that doesn't matter in regards to proton as ideally it should work as well as on Windows.

**Comment by Bitwolfies** [comment]
> On Steam Deck?
> 
> Fwiw it's the games audio engine that is a bit weird. You can make this happen with the native game on Windows, but you have to turn everything pretty overkill up compared to what your machine can handle, so it probably isn't a common problem there. The proton overhead or other differences seems to make it worse or more common. Or if you ntfs compress the games folder on Windows then it can happen too. But ofc that doesn't matter in regards to proton as ideally it should work as well as on Windows.

Yep, on deck it works fine now, are you saying it’s effectively tied to poor performance?

**Comment by Blisto91** [comment]
Interesting. I will check it out with GE here later.

> are you saying it’s effectively tied to poor performance?

I'm not exactly sure what causes it, maybe the Proton devs have more knowledge. But when i helped testing it i could make it happen on Windows when i moved the game to my HDD and set max graphics including render scale to 200% (Dunno if all steps were required). 
Tho i should note my PC is a bit of a potato hehe so at that point it ran at like at below 10fps during the cutscenes  so not really equivalent to Steam Deck as it actually runs it pretty decently on medium/high settings.

**Comment by ffrasisti** [comment]
Needs a fix on valve's proton still though. I wonder when the flickering grass will be fixed in either proton

**Comment by Bitwolfies** [comment]
> Needs a fix on valve's proton still though. I wonder when the flickering grass will be fixed in either proton

Supposedly not a proton thing, but a driver thing. If that’s the case you’ll need to wait for valve to rebase their mesa fork after it’s fixed and then system update the deck.

**Comment by Blisto91** [comment]
So i did try GE and it seemed to work well. But then afterwards i tried bleeding edge again and it seemed to work just as well. Even after a complete wipe and reinstall of the game.
Tho later i noticed there are still some audio that gets cut early. But i haven't seen any skipping back and forth or repeating, so not as bad as before. So i just got more confused lol.

**Comment by Bitwolfies** [comment]
> Tho later i noticed there are still some audio that gets cut early.

I’ve not seen any cuts in the few hours I’ve been playing since the patch.

**Comment by Blisto91** [comment]
They are small enough depending on graphical detail that they might not get noticed unless one look for them. Or maybe don't appear at all.
My testing has been replaying the intro cutscenes over and over again.
On highest graphical preset for me it cuts off a second too early when Prometheus describes some in the sea is as deep as Tartarus (can't remember the words)
Strangely it gets a bit more noticeable if i turn the present one step down. Here there is a place where Zeus will miss some of the words he speak at one point.
On lowest preset none of this happened.

But for some reason it's not as bad as i observed earlier, even on Bleeding Edge.

**Comment by ffrasisti** [comment]
So is it fixed on ge or not? i thought it was

**Comment by ffrasisti** [comment]
Did bleeding edge get another update?

**Comment by Blisto91** [comment]
> So is it fixed on ge or not? i thought it was

I dunno. Best is just to try for yourself and see if you notice any issues. It at least seems to have been reduced enough that people might not notice any problems.

> Did bleeding edge get another update?

It gets a update whenever there is anything to pull in from the subprojects. I don't know of anything related to this issue that has been updated.

**Comment by kakra** [comment]
> The issue apparently is fixed by this on a different game, AC Odyssey. something here affects _only the steam version_ of this game. Can someone from proton take a look please?

I can confirm that this very issue has been fixed for AC Odyssey before, so it's probably just an entry matching the exe file name somewhere in the code.

**Comment by Blisto91** [comment]
Yes it's the same hack applied to this game as they both use the same engine or close to.[ See here](https://github.com/ValveSoftware/wine/blob/30ed354c0d22d7646132731cd89111740f124f1d/dlls/ntdll/unix/loader.c#L2302)

It will even write ""HACK: AC Odyssey sync tweak on" in the proton log when active hehe

**Comment by kakra** [comment]
Thus, setting `WINE_SIMULATE_ASYNC_READ=1` should do the trick already for Proton versions not yet detecting this game.

Update: Well... unless it doesn't, the env var is new with that commit, too... :,(

**Comment by Blisto91** [comment]
The env var was added along with inserting the Fenyx Rising exe so it's only available on bleeding edge right now.

**Comment by kakra** [comment]
The uplay launcher and the game itself are unable to upload game saves to the cloud, so I'm unable to share my progress with my Steam Deck:

![20221222235417_1](https://user-images.githubusercontent.com/17684/209404944-91b01361-da4a-471e-8519-57b73895f910.jpg)
<img width="357" alt="Screenshot_20221222_235451" src="https://user-images.githubusercontent.com/17684/209404992-28a92d3f-bdf6-4b01-90ab-cd724ec686ba.png">

A similar problem seems to exist in other uplay titles such as AC Valhalla #6154.

**Comment by kakra** [comment]
> The env var was added along with inserting the Fenyx Rising exe so it's only available on bleeding edge right now.

@Blisto91 BTW: This is no longer true, looks like a few hours later, the fix was merged into Experimental and is available now.

**Comment by Blisto91** [comment]
Roger roger :+1: 

I have bisected the issue with the Ubisoft Connect overlay notification/popup staying stuck on screen and opened a upstream dxvk issue.

I haven't seen a issue with cloud saves from what i remember.

**Comment by Bitwolfies** [comment]
> Roger roger 👍
> 
> I have bisected the issue with the Ubisoft Connect overlay notification/popup staying stuck on screen and opened a upstream dxvk issue.
> 
> I haven't seen a issue with cloud saves from what i remember.

In terms of what I’ve seen from proton on Ubisoft connect, it can download saves fine, but it cannot upload them. No matter the Ubisoft game. Same behaviour applies to gog galaxy 2.0 so maybe it’s related. I also can confirm the stuck overlay.

**Comment by kakra** [comment]
> In terms of what I’ve seen from proton on Ubisoft connect, it can download saves fine, but it cannot upload them.

This is probably exactly what I am seeing because on the next start with the different PC, it asks me whether I want to use the local or the cloud save due to conflicts. That of course raises the question how the cloud save got there in the first place. But at least for my main PC, the cloud save is always the older one. So maybe it works in some of the times?

OTOH, downloads seem to be affected, too: The first time I started Valhalla on my Steam Deck, I already had some hours of game play on my main PC. But it started fresh. When I next started Valhalla on my main PC, it complained about a cloud save conflict - and I had to use the "older" local save to not lose my progress. After this, also the Steam Deck got my current progress. Since then, it seems to work for Valhalla but it doesn't for this game.

**Comment by Blisto91** [comment]
Roger. I haven't looked into that but will see if i can reproduce.

**Comment by kakra** [comment]
Another interesting observation: After quitting the game, cloud sync fails, and after skipping that error, it will tell me that my login has expired. Maybe there's some connection between those issues, and the login should not expire in the first place.

**Comment by kakra** [comment]
> After this, also the Steam Deck got my current progress. Since then, it seems to work for Valhalla but it doesn't for this game.

@Blisto91 Well, it stopped working for both games now: Cloud saves download but no longer upload, neither on Steam Deck nor on my main system. I'm not sure what changed.

**Update:** Out of the blue, it suddenly worked once, then it refused to work on repeated attempts. Seems the problem is pretty random, and more likely the upload fails instead of succeeding.

**Comment by Blisto91** [comment]
Sadness.
Maybe it's Ubisoft connect itself having problems connecting from time to time. Like when you install it the first time i sometimes have to restart like 5 times before it is able to find a update and not get stuck looking for updates forever

**Comment by kakra** [comment]
> Like when you install it the first time i sometimes have to restart like 5 times before it is able to find a update and not get stuck looking for updates forever

This seems to have been a different bug actually caused by multi-screen setups and DX init: https://github.com/ValveSoftware/Proton/issues/5882, at least since that fix, it never occurred for me again.

Still, I don't believe it has the same behavior on Windows, so something may be racing in wine and it doesn't in Windows - and that should be fixed.

**Comment by Blisto91** [comment]
I only have one monitor so probably not the same 😁
Haven't found why it happens yet

**Comment by kakra** [comment]
Retrospectively, I can remember that the Ubisoft launcher sometimes hung looking for updates on first launch, so I can confirm that observation. But I'm also seeing similar issues in other games, i.e., the Skyrim in-game extension shop sometimes times out during downloads or looking for updates, and trying again is usually enough to fix that. Also, games are not always expecting to wait forever on an HTTP download, and become essentially stuck because neither a timeout nor an error occurs, when visiting menus that do such downloads. OTOH, many games work just fine, so it's probably a specific implementation of using those functions which randomly fails. Maybe wine isn't detecting a specific (retryable?) error condition and just waits forever? And if that is not guarded by a wrapper to kill that thread after some timeout without progress, it will become stuck forever? I've seen the Ubisoft launcher hanging at random download progress values, so this seems plausible.

**Comment by ffrasisti** [comment]
Is the problem with the flickering grass solved?

**Comment by Blisto91** [comment]
No

**Comment by Blisto91** [comment]
The Ubisoft Overlay bug have received a fix in upstream dxvk.
The grass isn't fixed yet but a driver dev have been assigned to the issue.

**Comment by Bitwolfies** [comment]
Oh you have no idea how happy I am to hear the overlay is fixed. Absolutely crippled me trying to play far cry 5 with friends.

**Comment by Blisto91** [comment]
Grass flicker fix on amd radv has been merged in mesa main.

**Comment by libgradev** [comment]
Fails to launch here with recent kernels (6.4 / 6.3 series at least) - crashes after splash screen with an unhandled page fault. Unsure when this broke as haven't played it in a while.

Arch / AMD 6900XT / Mesa stable or current git.

Still launches fine with current LTS kernel (6.1.38).

**Comment by PaulM5406** [comment]
Failed to launch with proton experimental (2023/11/06) and proton 8.0-4 - here is the error:
![Screenshot from 2023-11-06 09-59-26](https://github.com/ValveSoftware/Proton/assets/33572994/7dcb27d3-1ebb-4b12-86ab-e8d0dcfd3528)
thank you for the help

**Comment by andrew-ld** [comment]
It seems that the audio randomly cuts out and jumps to the next dialogue with latest version of proton, it seems that to fix it you have to use an older version of proton-ge, I'm playing with proton-ge-7-43, there is no audio glitch or at least I didn't notice any in the first few hours of gameplay

**Comment by ffrasisti** [comment]
So is it fixed on the deck??

**Comment by andrew-ld** [comment]
yes, if you use proton ge 7 43 it works very good @ffrasisti
